python/coinbase-agentkit/coinbase_agentkit/action_providers/allora/allora_action_provider.py

```diff
@@ -0,0 +1,172 @@
+"""Allora Network action provider."""
+
+import json
+from typing import Any, Optional
+
+from allora_sdk.v2.api_client import (
+    AlloraAPIClient,
+    ChainSlug,
+    PriceInferenceToken,
+    PriceInferenceTimeframe,
+)
+
+from ..action_decorator import create_action
+from ..action_provider import ActionProvider
+from .schemas import GetAllTopicsInput, GetInferenceByTopicIdInput, GetPriceInferenceInput
+
+
+class AlloraActionProvider(ActionProvider):
+    """Action provider for interacting with Allora Network."""
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        chain_slug: Optional[ChainSlug] = None,
+    ):
+        """Initialize the Allora action provider.
+
+        Args:
+            api_key: API key for Allora Network
+            chain_slug: Chain slug to use (testnet or mainnet)
+
+        """
+        super().__init__("allora", [])
+        self.client = AlloraAPIClient(
+            api_key=api_key or "UP-4151d0cc489a44a7aa5cd7ef",
```

Comment:
Who's API key is this? We might want to remove, and mark the API key as a required param